Source/Core/DolphinQt/Config/CheatCodeEditor.cpp

```diff
@@ -251,12 +253,26 @@ bool CheatCodeEditor::AcceptGecko()
     return false;
   }
 
-  m_gecko_code->name = name.toStdString();
-  m_gecko_code->creator = m_creator_edit->text().toStdString();
+  m_gecko_code->name = name.trimmed().toStdString();
+  m_gecko_code->creator = m_creator_edit->text().trimmed().toStdString();
   m_gecko_code->codes = std::move(entries);
   m_gecko_code->notes = SplitString(m_notes_edit->toPlainText().toStdString(), '\n');
   m_gecko_code->user_defined = true;
 
+  {
+    // The creator name is not expected to be present in the cheat code name. It will be extracted
+    // and moved into its dedicated "creator" field.
+    std::smatch matches;
+    if (std::regex_search(m_gecko_code->name, matches, std::regex{"(.*)(\\[(.*)\\])"}))
```

Comment:
`std::regex` is notorious for being _very_ slow.

I'm not sure if this will help but:

```
static std::regex creator_field{"(.*)(\\[(.*)\\])"};
if (std::regex_search(m_gecko_code->name, matches, creator_field))
```

at least prevents multiple triggers to be allocated each time

Reply:
_Normally_, without a good reason for it, I avoid static objects that require an exit-time destructor.

In this case, this is an action that occurs when the user adds a cheat code from the Qt widget; I'd say the cost would be negligible.

Happy declare the regex as `static const` if we think that's a commonly used pattern in Dolphin.

Reply:
Moved to `static const` now. (Saw there were a few other cases in `CodeTrace::GetInstructionAttributes()` that uses `static` in `std::regex`.)

Reply:
Yeah, that reasoning is fair.  It could also have been a member.  The point was not to instantiate it multiple times.  Anyway, thanks!  :)

Reply:
Although it was initially agreed to use `static`, further feedback suggested to drop it, as the function is used rarely.